app/shared/ui-comment/src/desktopMain/kotlin/ui/comment/Turnstile.desktop.kt

```diff
@@ -0,0 +1,136 @@
+
+    private fun concatUrl(): String {
+        return "${url}&theme=${if (isDarkTheme) "dark" else "light"}"
+    }
+
+    fun initializeBrowser(): Component = runBlocking {
+        AniCefApp.suspendCoroutineOnCefContext {
+            val newClient = AniCefApp.createClient()
+                ?: throw IllegalStateException("AniCefApp should be initialized.")
+
+            client = newClient
+
+            val newBrowser = newClient.createBrowser(
+                concatUrl(),
+                CefRendering.DEFAULT,
+                true,
+                CefRequestContext.createContext { _, _, _, _, _, _, _ ->
+                    object : CefResourceRequestHandlerAdapter() {
+                        override fun onBeforeResourceLoad(
+                            browser: CefBrowser?,
+                            frame: CefFrame?,
+                            request: CefRequest?,
+                        ): Boolean {
+                            val requestUrl = request?.url
+                            println("Request URL: $requestUrl")
+                            if (requestUrl != null &&
+                                requestUrl.startsWith(TurnstileState.CALLBACK_INTERCEPTION_PREFIX)
+                            ) {
+                                tokenFlow.tryEmit(
+                                    requestUrl.substringAfter(
+                                        TurnstileState.CALLBACK_INTERCEPTION_PREFIX,
+                                    ),
+                                )
+                                return true
+                            }
+                            return super.onBeforeResourceLoad(browser, frame, request)
+                        }
+                    }
+                },
+            )
+
+            browser = newBrowser
+            newBrowser.setCloseAllowed()
+            newBrowser.uiComponent.apply { }
+        }
+    }
+
+    override fun reload() {
+        AniCefApp.runOnCefContext {
+            browser?.loadURL(concatUrl())
+        }
+    }
+
+    fun dispose() {
+        AniCefApp.blockOnCefContext {
+            browser?.close(true)
+            client?.dispose()
+        }
+    }
+}
+
+actual fun createTurnstileState(url: String): TurnstileState {
+    return DesktopTurnstileState(url)
+}
+
+@Composable
+actual fun ActualTurnstile(
+    state: TurnstileState,
+    constraints: Constraints,
+    modifier: Modifier,
+) {
+    check(state is DesktopTurnstileState)
+    val isDark = isSystemInDarkTheme()
+
+    SwingPanel(
+        background = Color.Transparent,
+        factory = {
+            state.isDarkTheme = isDark
+            state.initializeBrowser()
+        },
+        update = { component ->
+
+        },
+        modifier = modifier.fillMaxWidth().height(100.dp),
+    )
+
+    DisposableEffect(Unit) {
+        onDispose {
+            state.dispose()
+        }
+    }
+}
```

Comment:
Should not dispose here. You will make the `state` invalid and the caller may still use the same state instance